portfolio/src/main/java/com/google/sps/servlets/PageViewStatsServlet.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.data.ViewsEntity;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet for accumulating page view data */
+@WebServlet("/page-view-stats")
+public class PageViewStatsServlet extends HttpServlet {
+
+  /** Loads views from the datastore database and returns data from the last week. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Date date = new Date();
+    LocalDate currentDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+
+    LinkedHashMap<String, Long> views = new LinkedHashMap<>();
+    currentDate = currentDate.minusDays(6);
+    for (int i = 0; i < 7; i++) {
```

Comment:
This is slightly weird. You first subtract 6 days and then you have a for loop which goes forward again and adds one day each.
I suggest that you leave a variable called "currentDate" to really be the currentDate.
Then you can keep the for loop and simply calculate in the for loop what the dateOfInterest is and add the data for this date.